Docs: Add README for DatePicker and TimePicker Components (#70365)
WordPress/gutenberg: packages/components/src/date-time/time-picker/README.md

-   Required: No
-   Default: current time

### `onChange`: `( time: string ) => void`

The function called when a new time has been selected. It is passed the time as an ISO-formatted string.

-   Required: No

### `is12Hour`: `boolean`

Whether we use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed.

-   Required: No
-   Default: false

### `dateOrder`: `'dmy' | 'mdy' | 'ymd'`

The order of day, month, and year. This prop overrides the time format determined by `is12Hour` prop.

-   Required: No
-   Default: `'dmy'` (or `'mdy'` when `is12Hour` is `true`)

### `hideLabelFromVision`: `boolean`

Whether to visually hide field labels while keeping them accessible to screen readers.

-   Required: No
-   Default: false

## TimePicker.TimeInput

A standalone time input component. Values are passed as an object in 24-hour format (`{ hours: number, minutes: number }`).

### Usage

```jsx
import { useState } from 'react';
import { TimePicker } from '@wordpress/components';

const MyTimeInput = () => {
	const [ time, setTime ] = useState( { hours: 13, minutes: 30 } );

	return (
		<TimePicker.TimeInput
			value={ time }
			onChange={ setTime }
			label="Time"
		/>
	);

> };

Reviewer: ````suggestion
};
```
````

-   Default: current time

### `onChange`: `( time: string ) => void`

The function called when a new time has been selected. It is passed the time as an ISO-formatted string.

-   Required: No

### `is12Hour`: `boolean`

Whether we use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed.

-   Required: No
-   Default: false

### `dateOrder`: `'dmy' | 'mdy' | 'ymd'`

The order of day, month, and year. This prop overrides the time format determined by `is12Hour` prop.

-   Required: No
-   Default: `'dmy'` (or `'mdy'` when `is12Hour` is `true`)

### `hideLabelFromVision`: `boolean`

Whether to visually hide field labels while keeping them accessible to screen readers.

-   Required: No
-   Default: false

## TimePicker.TimeInput

A standalone time input component. Values are passed as an object in 24-hour format (`{ hours: number, minutes: number }`).

### Usage

​```jsx
import { useState } from 'react';
import { TimePicker } from '@wordpress/components';

const MyTimeInput = () => {
	const [ time, setTime ] = useState( { hours: 13, minutes: 30 } );

	return (
		<TimePicker.TimeInput
			value={ time }
			onChange={ setTime }
			label="Time"
		/>
	);
};

> ​```

Reviewer: Looks like there are zero-width space charaters used here, which may break formatting.

```suggestion
```jsx
import { useState } from 'react';
import { TimePicker } from '@wordpress/components';

const MyTimeInput = () => {
	const [ time, setTime ] = useState( { hours: 13, minutes: 30 } );

	return (
		<TimePicker.TimeInput
			value={ time }
			onChange={ setTime }
			label="Time"
		/>
	);
};
```
```

### Props

#### `value`: `{ hours: number, minutes: number }`

The time input value in 24-hour format.

-   Required: No

#### `defaultValue`: `{ hours: number, minutes: number }`

An optional default value for the control when used in uncontrolled mode. If left `undefined`, the current time will be used.

-   Required: No

#### `onChange`: `( time: { hours: number, minutes: number } ) => void`

Called when the time changes. Receives the new value as an object with `hours` and `minutes`.

-   Required: No

#### `is12Hour`: `boolean`

Whether to use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed.

-   Required: No

#### `label`: `string`

The label for the time input.

> -   Required: No

Reviewer: nit: missing trailing new line

```suggestion
-   Required: No

```

# TimePicker

TimePicker is a React component that renders a clock for time selection. It can be used independently or as part of the `DateTimePicker` component.

**Note:** `TimePicker` also exposes a compound sub-component, `TimePicker.TimeInput`, which can be used independently to render a time input field without the full picker UI.

## Usage

Render a TimePicker.

```jsx
import { useState } from 'react';
import { TimePicker } from '@wordpress/components';

const MyTimePicker = () => {
  const [ time, setTime ] = useState( new Date() );

  return (
    <TimePicker
      currentTime={ time }
      onChange={ ( newTime ) => setTime( newTime ) }
      is12Hour
    />
  );
};

> ```

Reviewer: Same here re. using tabs

## Props

The component accepts the following props:

### `currentTime`: `Date | string | number | null`

The current time at initialization. Optionally pass in a `null` value to specify no time is currently selected.

-   Required: No
-   Default: current time

### `onChange`: `( time: string ) => void`

The function called when a new time has been selected. It is passed the time as an ISO-formatted string.

-   Required: No

### `is12Hour`: `boolean`

Whether we use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed.

-   Required: No
-   Default: false

### `dateOrder`: `'dmy' | 'mdy' | 'ymd'`

> The order of day, month, and year.

Reviewer: Nit: blank line

```suggestion
The order of day, month, and year.

```

# TimePicker

> TimePicker is a React component that renders a clock for time selection. It can be used independently or as part of the `DateTimePicker` component.

Reviewer: ```suggestion
TimePicker is a React component that renders form inputs for time and date selection. It can be used independently or as part of the `DateTimePicker` component.
```

We should ideally apply the same update to the JSDoc description of `date-time/time-picker/index.tsx`

> **Note:** `TimePicker` also exposes a compound sub-component, `TimePicker.TimeInput`, which can be used independently to render a time input field without the full picker UI.

Reviewer: We could expand this section it with a brief usage example and the key props of `TimePicker.TimeInput`? The JSDoc in the source already has a good example we can adapt.

<details><summary>Suggestion</summary>

After the existing props section, add something like:

```md
## TimePicker.TimeInput

A standalone time input component. Values are passed as an object in 24-hour format (`{ hours: number, minutes: number }`).

### Usage

​```jsx
import { useState } from 'react';
import { TimePicker } from '@wordpress/components';

const MyTimeInput = () => {
	const [ time, setTime ] = useState( { hours: 13, minutes: 30 } );

	return (
		<TimePicker.TimeInput
			value={ time }
			onChange={ setTime }
			label="Time"
		/>
	);
};
​```

### Props

#### `value`: `{ hours: number, minutes: number }`

The time input value in 24-hour format.

-   Required: No

#### `defaultValue`: `{ hours: number, minutes: number }`

An optional default value for the control when used in uncontrolled mode. If left `undefined`, the current time will be used.

-   Required: No

#### `onChange`: `( time: { hours: number, minutes: number } ) => void`

Called when the time changes. Receives the new value as an object with `hours` and `minutes`.

-   Required: No

#### `is12Hour`: `boolean`

Whether to use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed.

-   Required: No

#### `label`: `string`

The label for the time input.

-   Required: No
```

</details>

Render a TimePicker.

```jsx
import { TimePicker } from '@wordpress/components';
import { useState } from '@wordpress/element';

const MyTimePicker = () => {

  const [ time, setTime ] = useState( new Date() );
  return (
    <TimePicker
      currentTime={ date }
      onChange={ ( newTime ) => setTime( newTime ) }
      is12Hour
    />
  );
};
```

## Props

The component accepts the following props:

### `currentTime`: `Date | string | number | null`

The current time at initialization. Optionally pass in a `null` value to specify no time is currently selected.

* Required: No
* Default: current time

### `onChange`: `( time: string ) => void`

The function called when a new time has been selected. It is passed the time as an ISO-formatted string.

* Required: No

### `is12Hour`: `boolean`

Whether we use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed and the time format is assumed to be `MM-DD-YYYY` (as opposed to the default format `DD-MM-YYYY`).

> 

Author: As far as I understand, is12Hour doesn’t actually control any date format or ordering. It only affects the time input UI—specifically whether an AM/PM selector is shown (12-hour format) or a 24-hour format is used.

The mention of MM-DD-YYYY vs DD-MM-YYYY here is misleading, since that relates to date field order, which is handled separately (e.g., via dateOrder). I’ll update the wording to reflect this more clearly.

Also noting **that this line was taken directly from the parent README**, but happy to refine it here for better accuracy. Please let me know if I’m missing anything.

Whether we use a 12-hour clock. With a 12-hour clock, an AM/PM widget is displayed.

-   Required: No
-   Default: false

### `dateOrder`: `'dmy' | 'mdy' | 'ymd'`

> The order of day, month, and year.

Reviewer: ```suggestion
The order of day, month, and year. This prop overrides the time format determined by `is12Hour` prop.
```
